src/osd/PrimaryLogPG.cc

```diff
@@ -1038,7 +1038,14 @@ void PrimaryLogPG::do_command(
 
     f->close_section();
   }
-
+  else if (prefix == "log") {
+    
+    f->open_object_section("op_log");
+    f->open_object_section("pg_log_t");
+    recovery_state.get_pg_log().get_log().dump(f.get());
```

Comment:
You probably need to be holding the pg lock here.

Reply:
Oh, this is PrimaryLogPG::do_command.  Check that the caller is already holding the lock, but I think it is.

Reply:
Yes, osd.cc will do the lock on pg before calling the do_command